common4j/src/main/com/microsoft/identity/common/java/commands/parameters/ATv2TokenCommandParameters.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.commands.parameters;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Command parameters class for ATv2 requests. Extends {@link InteractiveTokenCommandParameters} since ATv2
+ * can use the same set of parameters as a regular acquire token request. No additional parameters for ATv2 are
+ * required at this time, however.
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder(toBuilder = true)
+public class ATv2TokenCommandParameters extends InteractiveTokenCommandParameters {
```

Comment:
SLK is already being passed as an extra query parameter in the base token command parameters. I'd rather not extract it into its own field because that would mean we'd need to add another code path in acquirePrt to manually pass in slk in ATv2 scenarios to the authorize endpoint. This functionality is already present if SLK is in the extra query parameters field, I don't think we should ask oneauth to add it in 2 places within the same parameters object.

We could do it in a way where they only pass it in as a separate field in ATv2TokenCommandParameters, and we do some check in ATv2Activity to add it into the extra query parameter if it's not already there before sending the request through PrtController, but i feel this is adding more complexity that we probably don't need and is more or less still doing the parameter validation anyway.

Reply:
After some design changes, we've updated this to include slk as a direct parameter in this class.